axis.go

```diff
@@ -407,14 +418,63 @@ func (DefaultTicks) Ticks(min, max float64) (ticks []Tick) {
 		if val >= min && val <= max && !found {
 			ticks = append(ticks, Tick{Value: val})
 		}
-		if math.Nextafter(val, val+minorDelta) == val {
-			break
-		}
 		val += minorDelta
 	}
 	return
 }
 
+func removeDuplicates(elements []float64) []float64 {
+	// Uses a map to record duplicates as we find them.
+	encountered := map[float64]bool{}
+	result := []float64{}
+
+	for v := range elements {
+		if encountered[elements[v]] == false {
+			// Records this element as an encountered element.
+			encountered[elements[v]] = true
+			// Appends it to the result slice.
+			result = append(result, elements[v])
+		} // Otherwise, no elements are added as they are duplicates.
+	}
+	// Returns the new slice.
+	return result
+}
+
+func findRightLevelOfPrecision(elements []float64) []float64 {
+
+	var result []float64
+
+	for i := 1; i < 308; { //308 is the precision of the MaxFloat64 which we take as the maximum one here
+
+		var new_element float64
+
+		var v_mult float64
+
+		result = nil
+
+		for _, v := range elements {
+			v_mult = v * math.Pow10(i)
+			//As within 'if' statement v_mult is turned to integer, it cannot exceed the MaxInt64. If it does, another rounding technique is used.
+			if v_mult < (math.MaxInt64) {
+				new_element = float64(int(v_mult)) / math.Pow10(i)
+			} else {
+				new_element, _ = strconv.ParseFloat(strconv.FormatFloat(v, 'g', i, 64), 64)
+			}
+
+			result = append(result, new_element)
+		}
+		//checks whether the result array has duplicates. If not, the right level of precision is found, if it does, one more digit to the precision level is added and the check runs again.
+		if len(result) == len(removeDuplicates(result)) {
+			displayPrecision = i
+			break
```

Comment:
instead of `break`-ing, just `return result`.

Reply:
Changed that. Thanks!